apps/src/lib/script-editor/LevelTokenDetails.jsx

```diff
@@ -228,21 +272,44 @@ export class UnconnectedLevelTokenDetails extends Component {
         ))}
         {/* We don't currently support editing progression names here, but do
          * show the current progression if we have one. */}
-        {this.props.level.progression && (
+        {(this.props.level.progression || this.state.showBlankProgression) && (
           <div style={styles.progression}>
-            Progression Name: {this.props.level.progression}
+            <hr style={styles.divider} />
+            <span style={{...styles.levelFieldLabel}}>Progression name:</span>
+            <span style={styles.levelSelect}>
+              <input
+                type="text"
+                onChange={this.handleFieldChange.bind(this, 'progression')}
```

Comment:
same as above -- it looks like `handleFieldChange` takes `field` and `event` as its args, so this could be:

```
<input
  onChange={event => this.handleFieldChange('progression', event)}
/>
```